code/modules/research/experimentor/loot/_loot_definer.dm

```diff
@@ -0,0 +1,222 @@
+    var/loot_named
+    var/loot_described
+    // Keywords used to choose names for the second and third parts of the generated name.
+    // When defining new keywords, try to use ones that fit in the auto-generated descriptions.
+    var/list/loot_keywords
+    // Unique Names - If instantiated in the definition, will use one of the contained names instead of generating the third name.
+    var/list/unique_names
+
+// Returns a finished item from the stats supplied to the experimentor.
+/obj/experimentor/loot_definer/proc/define(var/stability_in, var/potency_in, var/base_name, var/rare_level, var/itemcategory)
+	stability = stability_in
+	potency = potency_in
+	raritylevel = rare_level
+	box_type = base_name
+	item_category = itemcategory
+	loot_named = TRUE
+	loot_described = TRUE
+	loot_keywords = new/list
+	unique_names = null
+
+    // determine what specific type (itemtype) the item is based on.
+	item_type = findItemType()
+    // define the item to be granted as loot.
+	loot_item = definitionByType()
+	// name and describe the item unless the definition says not to.
+	if(loot_named && loot_item != null)
+		loot_item.name = box_type + " [generateSecondName()][generateThirdName()]"
+	if(loot_described && loot_item != null)
+		loot_item.desc = generateDescription()
+	return loot_item
+
+
+///////////////////I. LOOT LIST////////////////////
+// Defines the loot lists for the experimentor
+// by category then by rarity.
+
+/obj/experimentor/loot_definer/proc/findItemType()
+    switch (item_category)
+        // Broad category list
+        if("Device")
+            // Lists by rarity.
+            if(raritylevel == RARITY_COMMON)
+                return pick("rapiddupe", "throwSmoke", "floofcannon", "teleport", "nothing")
+            if(raritylevel == RARITY_UNCOMMON)
+                return pick("teleport", "rapiddupe", "explode", "petSpray", "flash", "clean")
+            if(raritylevel == RARITY_RARE)
+                return pick("explode", "petSpray", "rapiddupe", "flash", "teleport")
+            if(raritylevel == RARITY_VERYRARE)
+                return pick("teleport", "explode", "rapiddupe")
+    return
+
+
+///////////////////// II. LOOT DEFINITIONS /////////////////////
+// This is what defines the object that actually gets returned.
+// Responsible for creating the object, populating its keywords
+// for naming and so on. set loot_named to FALSE if the item has
+// a unique name that you don't want to use the generator for.
+
+/obj/experimentor/loot_definer/proc/definitionByType()
+    // Switch here mostly for our benefit to break up categories.
+    // Try to keep your loot and device types in alphabetical order.
+    switch(item_category)
+        if("Device")
+            switch(item_type)
+                if("clean")
+                    loot_keywords.Add("destruction")
+                    return new/obj/item/discovered_tech(stability, potency, raritylevel, item_type)
```

Comment:
those all seem to return the same thing, so you have a lot of duplicate reutrns here. Why not simply have a single
return new/obj/item... at the end, after the ifs?

Reply:
I probably should do that, but this set up I did as an example; just because they all draw from the same object now doesn't mean they will always do so in that loot list.